File: wrappers/dotnet/indy-sdk-dotnet/PaymentsApi/UnknownPaymentMethodException.cs
namespace Hyperledger.Indy.PaymentsApi
{
    /// <summary>
    /// An unknown payment method was called.
    /// </summary>
    public class UnknownPaymentMethodException : IndyException
    {
        const string message = "An unknown payment method was called.";

        internal UnknownPaymentMethodException() : base(message, (int)ErrorCode.PaymentUnknownMethodError)
        {
        }
    }
}

File: wrappers/java/src/main/java/org/hyperledger/indy/sdk/payments/UnknownPaymentMethodException.java
package org.hyperledger.indy.sdk.payments;

import org.hyperledger.indy.sdk.ErrorCode;
import org.hyperledger.indy.sdk.IndyException;

/**
 * Exception thrown when unknown (e.g. unregistered) payment method was called
 */
public class UnknownPaymentMethodException extends IndyException {

    private static final long serialVersionUID = -8226688236266389417L;
    private static final String MESSAGE = "An unknown payment method was called";

    /**
     * Initializes a new {@link UnknownPaymentMethodException}
     */
    public UnknownPaymentMethodException() {
        super(MESSAGE, ErrorCode.UnknownPaymentMethod.value());
    }
}
